### PR TITLE
Query APIs

### PR DESCRIPTION
**Pull request by vladimir-moiseiev**
This PR adds missing API for QueryFactory and TopicFactory to request partial (filtered) data.

As minor changes, updating Spine proto package URL, fixing some existing tests.

### CODE REVIEW

File: test/common_factory_test.h [hunk 1]
```diff
@@ -51,20 +58,73 @@ class CommonFactoryTest : public ::testing::Test
...
-        command_factory_ = ActorRequestFactory::create(params).command_factory();
-        topic_factory_ = ActorRequestFactory::create(params).topic_factory();
+        query_factory_ = ActorRequestFactory::create(params_).query_factory();
+        command_factory_ = ActorRequestFactory::create(params_).command_factory();
+        topic_factory_ = ActorRequestFactory::create(params_).topic_factory();
+    }
+
+    std::vector<std::unique_ptr<ProjectId>> make_project_ids()
+    {
+        std::vector<std::unique_ptr<ProjectId>> project_ids;
```
Comment (illya-galashko) [inline test/common_factory_test.h]: Could be refactor 
std::vector<std::unique_ptr<ProjectId>> project_ids;
for(inti  = 0; i < 4; i++) {
auto project_id = std::make_unique_ptr<ProjectId>());
project_id->set_value(i);
project_ids.emplace_back(project_id);
}

File: include/spine/query_factory.h [hunk 2]
```diff
@@ -63,16 +70,132 @@ class QueryFactory
...
+     *
+     * @tparam T Protobuf Message type of a target entity.
+     * @tparam I Protobuf Message type of the entity IDs.
+     * @param ids         the entity IDs of interest.
+     * @param masks   the property paths for the \b FieldMask applied
+     *                    to each of results.
+     * @return an instance of \b Query formed according to the passed parameters.
+     */
+    template <typename T, typename = enable_param_if_protobuf_message<T>,
+            typename I, typename = enable_param_if_protobuf_message<I>
```
Comment (illya-galashko) [inline include/spine/query_factory.h]: indent

File: include/spine/util/any_utils.hpp [hunk 3]
```diff
@@ -49,7 +52,21 @@ namespace client {
  * @param message Protobuf message.
  * @return Protobuf Any packed from a message.
  */
-std::unique_ptr<::google::protobuf::Any> to_any(const ::google::protobuf::Message &message);
+inline std::unique_ptr<::google::protobuf::Any> to_any(const ::google::protobuf::Message &message)
+{
+    std::unique_ptr<::google::protobuf::Any> any { ::google::protobuf::Any::default_instance().New() };
```
Comment (illya-galashko) [inline include/spine/util/any_utils.hpp]: can be refactor with std::make_unique

File: include/spine/query_factory.h [hunk 4]
```diff
@@ -63,16 +70,132 @@ class QueryFactory
...
+    >
+    QueryPtr by_ids_with_masks(const std::vector<std::string> masks,
```
Comment (illya-galashko) [inline include/spine/query_factory.h]: Do we need to copy vector here?

File: include/spine/query_factory.h [hunk 5]
```diff
@@ -63,16 +70,132 @@ class QueryFactory
...
+     *
+     * Allows to set property paths for a \b FieldMask, applied to each of the query
+     * results. This processing is performed according to the
+     * [FieldMask specs](https://goo.gl/tW5wIU).
+     *
+     * In case the \b paths array contains entries inapplicable to the resulting entity
+     * (e.g. a \b path references a missing field), such invalid paths
+     * are silently ignored.
+     *
+     * @tparam T Protobuf Message type of a target entity.
```
Comment (armiol) [inline include/spine/query_factory.h]: Not sure, but I think that several `@tparam`s should be inter-aligned (like in [Java code](https://github.com/SpineEventEngine/core-java/blob/master/server/src/main/java/io/spine/server/model/HandlerMethod.java#L222)) and don't have periods at the end.

File: include/spine/query_factory.h [hunk 6]
```diff
@@ -63,16 +70,132 @@ class QueryFactory
...
+     * @param masks   the property paths for the \b FieldMask applied
+     *                    to each of results.
+     * @return an instance of \b Query formed according to the passed parameters.
```
Comment (armiol) [inline include/spine/query_factory.h]: In `@return` statements we are used to not have periods as well.

File: include/spine/util/target_utils.hpp [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+namespace spine {
+namespace client {
+
+
+template <typename T, typename = enable_param_if_protobuf_message<T>>
+std::unique_ptr<EntityFilters> make_entity_filters(const std::vector<std::unique_ptr<T>>& ids)
+{
+    std::unique_ptr<EntityIdFilter> entity_id_filter { EntityIdFilter::default_instance().New() };
+    for (auto &message : ids)
```
Comment (illya-galashko) [inline include/spine/util/target_utils.hpp]: please, move & to type

File: src/query_factory.cc [hunk 8]
```diff
@@ -36,25 +39,50 @@ QueryFactory::QueryFactory(std::unique_ptr<core::ActorContext>&& actor_context)
     actor_context_ = std::move(actor_context);
 }
 
-std::unique_ptr<Query> QueryFactory::all(const std::string& prefix, const std::string& type)
+QueryPtr QueryFactory::for_query(std::unique_ptr<Target>&& target)
```
Comment (illya-galashko) [inline src/query_factory.cc]: I believe && is uneccessary.

File: src/query_factory.cc [hunk 9]
```diff
@@ -36,25 +39,50 @@ QueryFactory::QueryFactory(std::unique_ptr<core::ActorContext>&& actor_context)
...
     query->set_allocated_target(target.release());
+    query->set_allocated_field_mask(clone(FieldMask::default_instance()));
 
     return query;
 }
 
+QueryPtr QueryFactory::for_query(std::unique_ptr<Target>&& target, std::unique_ptr<FieldMask> && field_mask)
+{
+    std::unique_ptr<Query> query = for_query(std::move(target));
+
```
Comment (illya-galashko) [inline src/query_factory.cc]: remove new lines according you your coding style

File: include/spine/util/any_utils.hpp (hunk 3 above)
Comment (vladimir-moiseiev) [reply include/spine/util/any_utils.hpp]: No it can't :)

**Pull request merged**